File: database_queries.js
const sqlite3 = require("sqlite3").verbose();

// Connect to the database
let db = new sqlite3.Database("log_data.db", sqlite3.OPEN_READWRITE, (err) => {
  if (err) {
    console.error(err.message);
  } else {
    console.log("Connected to the SQLite database.");
  }
});

let completed = false;

// Function to execute a single query
function executeQuery(sql) {
  return new Promise((resolve, reject) => {
    db.serialize(() => {
      let results = [];
      db.each(
        sql,
        (err, row) => {
          if (err) {
            reject(err.message);
            return;
          }
          results.push(row);
        },
        (err) => {
          if (err) {
            reject(err.message);
          } else {
            resolve(results);
          }
        }
      );
    });
  });
}

async function query(sql) {
  try {
    return await executeQuery(sql);
  } catch (er) {
    return { error: er.message };
  }
}

module.exports.query = query;

File: Log_CURD/DatabaseQueries.py
import sqlite3

db_file = r"C:\Users\hp\PycharmProjects\Analog\Log_CURD\log_data.db"


# Connect to the database
def connect_to_db():
    global db_file
    try:
        conn = sqlite3.connect(db_file)
        print("Connected to the SQLite database.")
        return conn
    except sqlite3.Error as err:
        print(err)
        return None


# Function to execute a single query
def execute_query(conn, sql):
    try:
        cursor = conn.cursor()
        cursor.execute(sql)
        results = cursor.fetchall()
        return results
    except sqlite3.Error as err:
        return {"error": str(err)}


# Main function to run a query
def query(sql):
    global db_file
    conn = connect_to_db()
    if conn is not None:
        results = execute_query(conn, sql)
        conn.close()
        return results
    else:
        return {"error": "Database connection failed"}
